module/VuFind/src/VuFind/View/Helper/Root/DisplayLanguageOptionFactory.php

```diff
@@ -65,10 +65,6 @@ public function __invoke(ContainerInterface $container, $requestedName,
         if (!empty($options)) {
             throw new \Exception('Unexpected options sent to factory.');
         }
-        // We want to construct a separate translator instance for this helper,
-        // since it configures different language/locale than the core shared
-        // instance!
-        $factory = new \VuFind\I18n\Translator\TranslatorFactory();
-        return new $requestedName($factory($container, Translator::class));
+        return new $requestedName($container->get(Translator::class));
```

Comment:
This apparently breaks the language selection.

Reply:
Can you elaborate on how it is broken? It works for me in my test environment (displaying the drop-down list with the text from native.ini). Is it possible you are seeing a cache-related problem? (Of course, I don't rule out the possibility that it's actually broken... but I want to be sure I test the right thing, since I thought I had this covered).

Reply:
After clearing the cache I'm seeing all choices in English.

Reply:
With dev they're in the native language.

Reply:
Oops, I should take my own advice: I was tricked by the cache! Thanks for catching this bug; I believe it is now fixed by c23ef01. Please let me know if it's sorted out on your end now, when time permits.

Reply:
Works great now. :)

Reply:
Great, thanks again! Merging now!